src/connector.js

```diff
@@ -0,0 +1,612 @@
+			suggestionsElement.style.display = 'block';
+		}
+	}
+}
+
+// Filters out dangerous URIs that can create XSS attacks such as `javascript:`.
+function filterProtocol( uri ) {
+
+	const isAbsolute = /^(https?|ftp|file|mailto|tel):/i.test( uri );
+	const isRelative = /^(\/|\.\/|\.\.\/)/.test( uri );
+
+	return isAbsolute || isRelative ? uri : '';
+}
+
+// Clear results list
+function clearResultList() {
+	resultListElement.textContent = "";
+}
+
+// Update results list
+function updateResultListState( newState ) {
+	resultListState = newState;
+	
+	if ( resultListState.isLoading ) {
+		if ( suggestionsElement ) {
+			suggestionsElement.style.display = 'none';
+        }
+	}
+	else {
+		clearResultList();
+		if( !resultListState.hasError && resultListState.hasResults ) {
+			resultListState.results.forEach( ( result, index ) => {
+				const sectionNode = document.createElement( "section" );
+				const highlightedExcerpt = HighlightUtils.highlightString( {
+					content: result.excerpt,
+					highlights: result.excerptHighlights,
+					openingDelimiter: '<strong>',
+					closingDelimiter: '</strong>',
+				} );
+				let itemId = `result-${index}`;
+				const resultDate = new Date( result.raw.date );
+
+				sectionNode.innerHTML = resultTemplateHTML
+					.replace( '%[result.clickUri]', filterProtocol( result.clickUri ) )
+					.replace( 'https://www.canada.ca', filterProtocol( result.clickUri ) ) // hack, invalid href are stripped
+					.replace( '%[result.title]', result.title )
+					.replace( '%[result.raw.author]', result.raw.author ? result.raw.author : '' )
+					.replace( '%[result.breadcrumb]', result.raw.displaynavlabel ? result.raw.displaynavlabel : result.printableUri )
+					.replace( '%[result.printableUri]', result.printableUri)
+					.replace( '%[short-date-en]', resultDate.getFullYear() + "-" + resultDate.getMonth() + 1 + "-" + resultDate.getDate() )
+					.replace( '%[short-date-fr]', resultDate.getFullYear() + "-" + resultDate.getMonth() + 1 + "-" + resultDate.getDate() )
+					.replace( '%[long-date-en]', monthsEn[ resultDate.getMonth() ] + " " + resultDate.getDate() + ", " + resultDate.getFullYear() )
+					.replace( '%[long-date-fr]', resultDate.getDate() + " " + monthsFr[ resultDate.getMonth() ] + " " + resultDate.getFullYear() )
+					.replace( '%[highlightedExcerpt]', highlightedExcerpt );
+				
+				const interactiveResult = buildInteractiveResult( headlessEngine, {
+					options: { result },
+				} );
+
+				let resultLink = sectionNode.querySelector( ".result-link" );
+				resultLink.onclick = ( e ) => { interactiveResult.select(); };
+				resultLink.oncontextmenu = ( e ) => { interactiveResult.select(); };
+				resultLink.onmousedown = ( e ) => { interactiveResult.select(); };
+				resultLink.onmouseup = ( e ) => { interactiveResult.select(); };
+				resultLink.ontouchstart = ( e ) => { interactiveResult.beginDelayedSelect(); };
+				resultLink.ontouchend = ( e ) => { interactiveResult.cancelPendingSelect(); };
```

Comment:
This can cause a WCAG issue. more testing is required to understand what is happening when those event are triggered.

Reply:
Bringing in @bblaisAtCoveo in the loop to better explain this and the implications.

Reply:
MANDATORY To provide explanations

Reply:
This code is necessary to build and push the proper Coveo usage analytics calls through the Coveo headless framework. A 'click' event is sent and it's ok to have the same call on all those events, ultimately only one API call will be pushed. The 'click' event on a result is an important event used to improve relevancy of the results over time, and to create reports that tracks usage on your site.